client/reader/lib/author-name-blacklist/index.ts

```diff
@@ -12,7 +11,7 @@ const authorNameBlacklist = [ 'admin' ];
  * @param {string} authorName Author name
  * @returns {boolean} True if blacklisted
  */
-export const isAuthorNameBlacklisted = authorName => {
+export const isAuthorNameBlacklisted = ( authorName: string ): boolean => {
```

Comment:
Question: are these redundant if TypeScript has inferred the types itself?

Reply:
I think it's better to err on the side of explicitness for function type signatures. 

For this case, it's actually possible to infer authorName's type as `string | null | undefined`.

Thoughts, @Automattic/type-review?

Reply:
setting `authorName: string` makes a lot of sense because it communicates to the callers what is expected.

TypeScript should be able to infer the return type and in this case it should know it's boolean: it does in my IDEs. Strangely Visual Studio Code shows it as returning `boolean` but PHPStorm shows it returning `(false | boolean)`. Both are technically true but maybe redundant. The distinction I think is that it can carry the `false` out specifically to other places where it might impose further constraints.

That being said I don't think this input type is right. The way it's typed as `string` makes me think that it could never be missing and therefore we return `false` when the name is an empty string, but that doesn't make sense. I think the function is guarding against not getting a string and so we should use `( authorName?: string )` or `( authorName: string = '' )`

Reply:
there's an element here that's going to be funny for some time as we transition. although we're adding types here those constraints aren't being held inside of our `.js` files, so the places where we call `isAuthorNameBlacklisted()` aren't being flagged for sending the wrong data.

this isn't a reason _not_ to type these leaf functions, but it is a good reason to leave our existing type allowances in. that is, although if we were compiling with TypeScript it would force us to pass a `string` whenever we call this, we don't have the guarantee with Babel doing the compiling. it's kind of a pickle. the type assertion should allow us to remove the `if ( ! authorName )` (unless the behavior really is intending to check against empty user names, which would be redundant) but the possibility of getting an undefined author name means we have to leave it in.

this kind of change might be a lot easier if we start rewriting consumer sites rather than edge-nodes. of course, that involves long rabbit-holes.

---

for funnies, her is how I might collapse this function in a rewrite. it leaves enough of a guard to catch cases where the author name isn't sent but still maintains the types you want to give it and end up with.

```ts
export const isAuthorNameBlacklisted = ( authorName: string ) =>
	!! authorName && 'admin' === authorName.toLowerCase();
```

it started out without the `!!` then TypeScript reminded me that the return value would then have been `( "" | boolean )` vs just `boolean`